src/library/masq.js

```diff
@@ -414,30 +440,62 @@ class Masq {
         try {
           if (app) {
             const privateProfile = await this.getProfile(this.profileId)
-            await sendAuthorized(peer, app.id, app.appDEK, privateProfile.username, privateProfile.image, app.appNonce)
+            const res = await this.sendAuthorized(peer, app.id, app.appDEK, privateProfile.username, privateProfile.image, app.appNonce)
+
+            resolve(res)
           } else {
-            await sendNotAuthorized(peer)
+            const res = await this.sendNotAuthorized(peer)
+            resolve(res)
           }
         } catch (e) {
           await this._closeUserAppConnection()
-          return reject(new MasqError(MasqError.DISCONNECTED_DURING_LOGIN))
+          return reject(e)
         }
+      })
+    })
+  }
 
-        peer.once('data', async (data) => {
-          const json = await decrypt(this.key, JSON.parse(data), 'base64')
+  async sendAuthorized (peer, userAppDbId, userAppDEK, username, profileImage, userAppNonce) {
```

Comment:
don't you find easier to let the function `sendAuthorized` just the task of sending the authorized message, like previously ? And define another function to read the peer response

Reply:
ok, done